src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -238,7 +272,91 @@ private void validateSsl() {
     }
   }
 
+  private void validateConnection() {
+    try {
+      if (!client.ping(RequestOptions.DEFAULT)) {
+        throw new IOException("Cannot connect to Elasticsearch.");
+      }
+    } catch (IOException e) {
+      String errorMessage = String.format(
+          "Could not connect to Elasticsearch due to %s",
+          e.getMessage()
+      );
+      addErrorMessage(CONNECTION_URL_CONFIG, errorMessage);
+
+      if (config.isAuthenticatedConnection()) {
+        errorMessage = String.format(
+            "Could not authenticate the user. Check the '%s' and '%s'. Error message: %s.",
+            CONNECTION_USERNAME_CONFIG,
+            CONNECTION_PASSWORD_CONFIG,
+            e.getMessage()
+        );
+        addErrorMessage(CONNECTION_USERNAME_CONFIG, errorMessage);
+        addErrorMessage(CONNECTION_PASSWORD_CONFIG, errorMessage);
+      }
+
+      if (config.isSslEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your SSL settings. Error message: %s",
+            e.getMessage()
+        );
+
+        addErrorMessage(SECURITY_PROTOCOL_CONFIG, errorMessage);
+      }
+
+      if (config.isKerberosEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your Kerberos settings. Error message: %s",
+            e.getMessage()
+        );
+
+        addErrorMessage(KERBEROS_PRINCIPAL_CONFIG, errorMessage);
+        addErrorMessage(KERBEROS_KEYTAB_PATH_CONFIG, errorMessage);
+      }
+
+      if (config.isBasicProxyConfigured()) {
```

Comment:
If we can't ping the ES, doesn't necessarily mean it's because of proxy settings. Maybe we can do some heuristics here, e.g. inspect exception, whether it has HTTP 407 - that means proxy authentication had failed.

Reply:
`ping` only returns `boolean`. We don't actually get exposed to the response codes so unfortunately we can't determine the exact root cause.